### PR TITLE
Adding tag partial fill when order is partially filled

### LINKED ISSUE #155: Surplus + Status label improvements
**Description**
(reported by Anxo)
Surplus breaks a bit with small numbers, and also we don’t handle too well partial executions: The state and surplus of this one could be improved.

We can show the surplus, and the status is something like PARTIAL FILLED

**How to Reproduce**
1. Go to https://explorer.cow.fi/tx/0x8bb4f9644bdf19684d66c55133ded17c198e36069a502f903311eb07ed978145?tab=orders
image

### PR DESCRIPTION
**Pull request by henrypalacios**
# Summary

Closes #155 

**Proposal**
1. Increase column width to fit a surplus with hundredth.
. ~~Use a tag orange label when a order is partially fill~~
image

2. Create a new status label `Partially filled` when order is **expired** and **partially filled**.
image

## To Test
- **Label** `partiall fill` positioned at the **bottom**
  [Transaction Details View](https://pr172--explorer.review.gnosisdev.com/tx/0x8bb4f9644bdf19684d66c55133ded17c198e36069a502f903311eb07ed978145?tab=orders)

- **Label** `partiall fill` positioned at the **right**
  [Order Details View](https://pr172--explorer.review.gnosisdev.com/orders/0x93299f3754e51938baae1946583382f9906092213cf8042b81eb877c8923fb67e63a13eedd01b624958acfe32145298788a7a7ba624ae706)
  
- [Storybook Status Label](https://pr172--explorer.review.gnosisdev.com/storybook/?path=/story/orders-statuslabel--expired-partially-filled)

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @henrypalacios , very-very nice changes!

Maybe a nitpick, but I'd start the 'Partial fill' part with the capital letter. 
Part filled

And I'd write 'filled' instead of 'fill', but I'm not stick to this point.

**Review by alongoni — APPROVED** [review]
Looks very nice!

I've the same doubt than Elena, maybe it's "Partial filled"

**Comment by henrypalacios** [comment]
@alongoni 
I would prefer the label without uppercase
 Screenshot from 2022-08-04_09-39
:point_up: :see_no_evil:

File: yarn.lock [hunk 1]
```diff
@@ -19279,23 +19267,7 @@ style-to-object@0.3.0, style-to-object@^0.3.0:
   dependencies:
     inline-style-parser "0.1.1"
 
-styled-components@^5.0.0:
-  version "5.3.5"
```
Comment (henrypalacios) [inline yarn.lock]: A.K.A trouble maker :laughing:!

File: src/components/orders/StatusLabel/index.tsx [hunk 2]
```diff
@@ -55,15 +104,15 @@ type ShimmingProps = {
   shimming?: boolean
 }
 
-const Label = styled.div<DisplayProps & ShimmingProps>`
+const Label = styled.div<DisplayProps & ShimmingProps & PartiallyTagProps>`
   font-weight: ${({ theme }): string => theme.fontBold};
-  text-transform: capitalize;
```
Comment (henrypalacios) [inline src/components/orders/StatusLabel/index.tsx:60]: @alongoni 
I have not been able to get the formatting to work for text with: 
```css
&::first-letter {
    text-transformation: uppercase;
}
```

So I've had to go old school with JavaScript [L182](https://github.com/cowprotocol/explorer/blob/155-fix-row-surplus/src/components/orders/StatusLabel/index.tsx#L182)

**Review by ramirotw — APPROVED** [review]

File: src/components/orders/StatusLabel/index.tsx [hunk 3]
```diff
@@ -33,6 +41,18 @@ function setStatusColors({ theme, status }: { theme: DefaultTheme; status: Order
       text = theme.labelTextOpen
       background = theme.labelBgOpen
       break
+    case 'partially filled':
+      return css`
+        background: linear-gradient(270deg, rgba(0, 216, 151, 0.1) 0%, rgba(217, 109, 73, 0.1) 100%);
```
Comment (alfetopito) [inline src/components/orders/StatusLabel/index.tsx:46]: I'm not sold on the gradient.
I rather keep it solid

**Review by alfetopito — COMMENTED** [review]
Comment regarding the gradient.

A note for the reviewers, you can use the [storybook](https://pr172--explorer.review.gnosisdev.com/storybook/?path=%2Fstory%2Forders-statuslabel--open-partially-filled) to view all states that cannot be easily reproduced

Also the partial fill color does look odd with the cancelled label
Screen Shot 2022-08-08 at 11 13 49

**Comment by henrypalacios** [comment]
@elena-zh I have confirmed see the changes suggested by @alfetopito  in the **storybook**.

Let me know if you were able to test partial orders.

File: src/components/orders/StatusLabel/index.tsx (hunk 3 above)
Comment (henrypalacios) [reply src/components/orders/StatusLabel/index.tsx:46]: Done! 
image

**Comment by elena-zh** [comment]
Hi Henry. 
No, I was not able to get partially filled orders in Open/Cancelling/Cancelled states. Maybe I should create orders with great amounts or so. but all my orders in 3 networks were always filled. ([Here ](https://pr172--explorer.review.gnosisdev.com/rinkeby/address/0xde1e8eedd42cd284a3052d987502d0b24c633fee) is account details which was white-listed for partially-filled orders)
I will give another try tomorrow. 

I still have concerns with this gradient: looks like a dirty-green color to me (personally)
image

Another issue I have faced in this PR is that 'inverse' arrows became too big
huge

File: src/components/orders/StatusLabel/index.tsx (hunk 3 above)
Comment (alfetopito) [reply src/components/orders/StatusLabel/index.tsx:46]: Oh, I missed this gradient!
I actually saw the one Elena mentioned below and thought that was the only one.

Could you make the `Partially filled` status solid as well?

**Comment by elena-zh** [comment]
Hey @henrypalacios , still arrows are huge ones on order details page 
image

And I can see this ugly scroll on transaction OR order derails pages on certain screen resolutions
image, but I reported a separate issue #176 for this. 

Also, here is testing link for **Cancelled+Partially filled order** : https://pr172--explorer.review.gnosisdev.com/orders/0x4a3927813f512fcc6ba710ee44d2a9204d21e997c5c122a60d1d4a2106fdd848042b32ac6b453485e357938bdc38e0340d4b927661c2639b

Just a question: can we use a status that i implemented in Storybook for order details page? Due to it seems to me that showing the status in 1 line makes it too long+ different fonts size is too much visible.. @alongoni , @alfetopito , WDYT?
diff font
image

**Comment by henrypalacios** [comment]
@elena-zh I Fixed the Icon,

@alongoni , let's look at the scroll bar problem

Regarding the last point, I did it in column mode because there was enough space to show it at right. But if it is the team's preference we can always use the row version.
image

**Comment by anxolin** [comment]
Hi, I prepared some feedback for the label. The one i really like is the Expired and Partially Fillable. The others, look a bit strange to me. 

https://docs.google.com/document/d/1WXjxJHUqljOdhLBoMjpluqeFTFYkR6TX6XM3DsbpccI/edit#

File: src/components/orders/StatusLabel/index.tsx [hunk 4]
```diff
@@ -8,13 +8,24 @@ import {
...
+  faCircleHalfStroke,
 } from '@fortawesome/free-solid-svg-icons'
 
 import { OrderStatus } from 'api/operator'
 
-type DisplayProps = { status: OrderStatus }
+type CustomOrderStatus = OrderStatus | 'partially filled'
+type DisplayProps = { status: CustomOrderStatus }
+function canBePartiallyFilled(status: string): status is OrderStatus {
+  return ['open', 'cancelled', 'cancelling'].includes(status) // excluding 'expired' because it will have own statusLabel
```
Comment (alfetopito) [inline src/components/orders/StatusLabel/index.tsx:19]: As per discussion in the doc, should exclude `cancelled` as well

File: src/components/orders/StatusLabel/index.tsx [hunk 5]
```diff
@@ -110,19 +174,21 @@ function StatusIcon({ status }: DisplayProps): JSX.Element {
...
-export type Props = DisplayProps & { partiallyFilled: boolean }
+export type Props = { status: OrderStatus; partiallyFilled: boolean; partialTagPosition?: PartiallyTagPosition }
 
 export function StatusLabel(props: Props): JSX.Element {
-  const { status, partiallyFilled } = props
+  const { status, partiallyFilled: partiallyFilled, partialTagPosition = 'bottom' } = props
   const shimming = status === 'signing' || status === 'cancelling'
+  const isExpired = status === 'expired'
+  const tagPartiallyFilled = partiallyFilled && canBePartiallyFilled(status)
+  const _status = isExpired && partiallyFilled ? 'partially filled' : status
```
Comment (alfetopito) [inline src/components/orders/StatusLabel/index.tsx:184]: Should add `cancelled` status here too

**Comment by elena-zh** [comment]
Hey @henrypalacios , seems that Expired+partially filled status is broken in the Storybook
image

**Comment by elena-zh** [comment]
> Hey @henrypalacios , seems that Expired+partially filled status is broken in the Storybook image

No longer relevant

**Comment by henrypalacios** [comment]
@alfetopito according to the suggestions, when an order is partially filled and has been '**cancelled**' or has '**expired**' it will show the tag :point_down: 
image

On another line, we have '**open**' or '**cancelling**' which when partially filled shows the percentage as suggested by @anxolin 
1. Tx view, User Orders view
image
image

2. Order details view
image
image

Finally @elena-zh, let me know if you have found a way to test the different states.

File: src/components/orders/StatusLabel/index.tsx [hunk 6]
```diff
@@ -110,19 +177,35 @@ function StatusIcon({ status }: DisplayProps): JSX.Element {
...
-        <StatusIcon status={status} />
-        {status}
+    <Wrapper
+      partialFill={tagPartiallyFilled}
+      tagPosition={partialTagPosition}
+      filledPercentage={formattedPercentage || 'Partially filled'}
+    >
+      <Label status={_status} shimming={shimming} partialFill={tagPartiallyFilled} tagPosition={partialTagPosition}>
+        <StatusIcon status={_status} />
+        {_status.charAt(0).toUpperCase() + _status.slice(1)}
```
Comment (alfetopito) [inline src/components/orders/StatusLabel/index.tsx:207]: Nitpick: Any reason not to use `_status[0]` instead?

Also, this doesn't feel right. Could you create an util function for capitalizing the string?
I'm sure we created them before, just not sure in which repo it was.

**Review by alfetopito — APPROVED** [review]
I have some small nitpicks regarding the code and one regarding the UI.

Feels like the text is too close to border. What about adding a little bit of padding to the top/bottom?
Screen Shot 2022-08-18 at 10 45 42

**Comment by elena-zh** [comment]
Hey @henrypalacios , great changes!
Agree with @alfetopito that it would be great to add an extra-padding here 
image

I have been able to get a partially filled order (finally!!!), and check its 'Open' state. However, I have found an issue with Cowswap UI side #941: partially  filled order is displayed as 'successfully filled' on the UI, so I can't test 'cancelling' state until the issue is fixed. 

`Cancelled + Partially filled`, `Expired + Partially filled` statuses LGTM.

File: src/components/orders/StatusLabel/index.tsx (hunk 6 above)
Comment (henrypalacios) [reply src/components/orders/StatusLabel/index.tsx]: I agree, I reused `capitalize` function.

**Pull request merged**